uaclient/api/u/pro/token_info/v1.py

```diff
@@ -0,0 +1,293 @@
+        self.name = name
+        self.auto_enabled = auto_enabled
+
+
+class TokenInfoResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("account", AccountInfo),
+        Field("contract", ContractInfo),
+        Field("services", data_list(ServiceInfo)),
+    ]
+
+    def __init__(
+        self,
+        account: AccountInfo,
+        contract: ContractInfo,
+        services: List[ServiceInfo],
+    ):
+        self.account = account
+        self.contract = contract
+        self.services = services
+
+
+def _get_application_entitlement_information(
+    entitlements: List[Dict[str, Any]], entitlement_name: str
+) -> Dict[str, Any]:
+    """Extract information from the entitlements array."""
+    for entitlement in entitlements:
+        if entitlement.get("type") == entitlement_name:
+            return {
+                "entitled": entitlement.get("entitled"),
+                "auto_enabled": (
+                    entitlement.get("obligations", {}).get("enableByDefault")
+                ),
+            }
+    return {"entitled": False, "auto_enabled": False}
+
+
+def get_token_info(options: TokenInfoOptions) -> TokenInfoResult:
+    return _get_token_info(options=options, cfg=UAConfig())
+
+
+def _get_token_info(
+    options: TokenInfoOptions, cfg: UAConfig
+) -> TokenInfoResult:
+    token = options.token
+    try:
+        contract_information = get_contract_information(cfg, token)
+    except exceptions.ContractAPIError as e:
+        if hasattr(e, "code") and e.code == 401:
+            raise exceptions.AttachInvalidTokenError()
+        raise e
+
+    contract_info = contract_information.get("contractInfo", {})
+
+    account_info = contract_information.get("accountInfo", {})
+    account = AccountInfo(
+        id=account_info.get("id", ""), name=account_info.get("name", "")
+    )
+
+    # Check contract expiration
+    effective, expires = None, None
+    now = datetime.now(timezone.utc)
+    if contract_info.get("effectiveTo"):
+        expiration_datetime = contract_info.get("effectiveTo")
+        expires = expiration_datetime
+        delta = expiration_datetime - now
+        if delta.total_seconds() <= 0:
+            raise exceptions.AttachForbiddenExpired(
+                contract_id=contract_info.get("id", ""),
+                date=expiration_datetime.strftime(ATTACH_FAIL_DATE_FORMAT),
+            )
+    if contract_info.get("effectiveFrom"):
+        effective_datetime = contract_info.get("effectiveFrom")
+        effective = effective_datetime
+        delta = now - effective_datetime
+        if delta.total_seconds() <= 0:
+            raise exceptions.AttachForbiddenNotYet(
+                contract_id=contract_info.get("id", ""),
+                date=effective_datetime.strftime(ATTACH_FAIL_DATE_FORMAT),
+            )
+
+    contract = ContractInfo(
+        id=contract_info.get("id", ""),
+        name=contract_info.get("name", ""),
+        effective=effective,
+        expires=expires,
```

Comment:
I think the API should return the datetime object directly for those fields, not just return the string representation